source/guide_prosody.rst

```diff
@@ -0,0 +1,309 @@
+
+Configure luarocks
+------------------
+
+Prosody is written in ``lua`` and has some dependencies which we install with the package manager ``luarocks``. In order to run the installed packages we have to adapt the path:
+
+::
+
+ PATH=$HOME/.luarocks/bin:$PATH
+ 
+ export $PATH
+
+Additionally we need to provide the paths ``LUA_PATH`` and ``LUA_CPATH`` to use ``luarocks`` accordingly:
+
+::
+
+ [isabell@stardust ~]$ echo "$(luarocks path)" >> ~/.bash_profile \
+ && source ~/.bash_profile
+ [isabell@stardust ~]$
+
+Install lua-dependencies
+------------------------
+
+The following dependencies_ (``luasocket``, ``luaexpat`` and ``luafilesystem``) are required:
+
+::
+
+ [isabell@stardust ~]$ luarocks install luasocket --local && \
+ luarocks install luaexpat --local EXPAT_BINDIR="/usr/bin" EXPAT_INCDIR="/usr/include/" EXPAT_LIBDIR="/usr/lib64" && \
+ luarocks install luafilesystem --local && \
+ luarocks install luasec --local OPENSSL_BINDIR="/usr/bin" OPENSSL_INCDIR="/usr/include" OPENSSL_LIBDIR="/usr/lib64"
+ [...]
+ luasocket [...] is now built and installed in [...]
+ luaexpat [...] is now built and installed in [...]
+ luafilesystem [...] is now built and installed in [...]
+ luasec [...] is now built and installed in [...]
+ [isabell@stardust ~]$
+
+.. note:: The variables ``*_BINDIR`` ``*__INCDIR`` and ``*_LIBDIR`` are necessary for correct linking the associated library because CentOS uses a different layout for those files than luarocks expects!
+
+Further optional ones (``luadbi-mysql``, ``lua-zlib`` and ``luaevent``) can be installed with these commands:
+
+::
+
+ [isabell@stardust ~]$ luarocks install luadbi-mysql --local MYSQL_BINDIR="/usr/bin" MYSQL_INCDIR="/usr/include/mysql" MYSQL_LIBDIR="/usr/lib64" && \
+ luarocks install lua-zlib --local && \
+ luarocks install luaevent --local
+ [...]
+ luadbi [...] is now built and installed in [...]
+ luadbi-mysql [...] is now built and installed in [...]
+ lua-zlib [...] is now built and installed in [...]
+ luaevent [...] is now built and installed in [...]
+ [isabell@stardust ~]$
+
+To list the installed packages with their versions use the command ``luarocks list``.
+
+Set c99 compilation flag
+------------------------
+In order to built prosody we have to activate C99-mode as global setting for all packages compiled via luarocks ``~/.luarocks/config-5.1.lua``.
+
+::
+
+ variables = {
+ CC= "gcc -std=c99"
+ }
+
+.. note:: This is a temporary workaround since luarocks not allowing setting this flag via ``.rockspec``-file.
+
+MySQL-database
+--------------
+
+Create a separate db for prosody data:
+
+::
+
+ [isabell@stardust ~]$ mysql -e "CREATE DATABASE ${USER}_prosody CHARACTER SET utf8"
+ [isabell@stardust ~]$
+
+Installation
+============
+
+Install prosody with an (currently unofficial) ``.rockspec``-file:
+
+::
+
+ [isabell@stardust ~]$ luarocks install --server=http://luarocks.org/manifests/fapsi prosody --local
```

Comment:
This is your personal luarocks manifest, right? Guides should use the official sources, if at all possible. Is there a way to install prosody using the official sources only? :/

Reply:
I thought that it would be prettier to use luarocks for the prosody installation step too, because it's easier to maintain and update, but you're right with using the official sources. Since luarocks has the mentioned problems the guide looks clearer now by building from the sources.
Another not quite related question: Since prosody 0.12 will only support Lua 5.2, are there any plans to provide it on CentOS 7?